quiche/src/pmtud.rs

```diff
@@ -1,92 +1,193 @@
+        self.current_mtu
     }
 
     /// Selects path MTU probe based on the binary search algorithm.
     ///
     /// Based on the Optimistic Binary algorithm defined in:
     /// Ref: <https://www.hb.fh-muenster.de/opus4/frontdoor/deliver/index/docId/14965/file/dplpmtudQuicPaper.pdf>
     pub fn update_probe_size(&mut self) {
-        self.probe = self.cur_size + ((self.probe - self.cur_size) / 2);
+        self.probe_size =
+            self.current_mtu + ((self.probe_size - self.current_mtu) / 2);
     }
 
     /// Updates probe value when the Path MTU Discovery probe is lost.
     pub fn pmtu_probe_lost(&mut self) {
         self.update_probe_size();
-        self.should_probe(true);
+        self.set_should_probe(true);
     }
 }
 
 impl std::fmt::Debug for Pmtud {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
-        write!(f, "current={:?} ", self.cur_size)?;
-        write!(f, "probe_size={:?} ", self.probe)?;
-        write!(f, "continue_probing={:?} ", self.next_size)?;
-        write!(f, "enable={:?} ", self.enable)?;
+        write!(f, "current_mtu={:?} ", self.current_mtu)?;
+        write!(f, "probe_size={:?} ", self.probe_size)?;
+        write!(f, "should_probe={:?} ", self.should_probe)?;
+        write!(f, "enabled={:?} ", self.enabled)?;
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn pmtud_initial_state() {
+        let mut pmtud = Pmtud::new(1200);
+
+        assert_eq!(pmtud.get_current_mtu(), 1200);
+        assert_eq!(pmtud.get_probe_size(), 0);
+        assert!(!pmtud.get_should_probe());
+        assert!(!pmtud.is_enabled());
+    }
+
+    #[test]
+    fn pmtud_enable_disable() {
+        let mut pmtud = Pmtud::new(1200);
+
+        pmtud.enable(true);
+        assert!(pmtud.is_enabled());
+
+        pmtud.enable(false);
+        assert!(!pmtud.is_enabled());
+    }
+
+    #[test]
+    fn pmtud_probe_flag_management() {
+        let mut pmtud = Pmtud::new(1200);
+
+        // Initially should not probe
+        assert!(!pmtud.get_should_probe());
+
+        // Enable probing
+        pmtud.set_should_probe(true);
+        assert!(pmtud.get_should_probe());
+
+        // Disable probing
+        pmtud.set_should_probe(false);
+        assert!(!pmtud.get_should_probe());
+    }
+
+    #[test]
+    fn pmtud_binary_search_algorithm() {
+        let mut pmtud = Pmtud::new(1200);
+
+        // Set initial probe size to 1500
+        pmtud.set_probe_size(1500);
+        assert_eq!(pmtud.get_probe_size(), 1500);
+
+        // Simulate probe loss - should update to midpoint
+        pmtud.update_probe_size();
+        // Expected: 1200 + ((1500 - 1200) / 2) = 1200 + 150 = 1350
+        assert_eq!(pmtud.get_probe_size(), 1350);
+
+        // Another probe loss
+        pmtud.update_probe_size();
+        // Expected: 1200 + ((1350 - 1200) / 2) = 1200 + 75 = 1275
+        assert_eq!(pmtud.get_probe_size(), 1275);
```

Comment:
It would be good to cover the `1201` to `1200` transition in a test.

Reply:
Done